### ISSUE TITLE
Overflow on longer strings

### ISSUE THREAD

**Issue by arminfriedl** [jcs090218/flx-rs#2]
Rust panics with
`thread 'test_nok' panicked at 'attempt to multiply with overflow', src/search.rs:256:40`
on longer strings. Maybe somewhat surprisingly, both the query as well as the search string seem to contribute to this panic.

```rust
#[test]
fn test_ok() {// base, works
    let s = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890";
    score(s, "ABCDE");
}

#[test]
fn test_ok2() {// s increased by one, query reduced by one, still works
    let s = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890A";
    score(s, "ABCD");
}

#[test]
fn test_panic() {// s increased by one, query left as is, panics
    let s = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890A";
    score(s, "ABCDE");
}
```

I'm not sure whether only short strings are supported or this is a legit bug.

**Comment by jcs090218** [comment]
I have widen the character bit type from `u8` to `u32`. Hope it works!

Sorry I am very new to Rust, so I may did something weird in my code.

> I'm not sure whether only short strings are supported or this is a legit bug.

This library is port from Emacs [flx](https://github.com/lewang/flx) library. I would consider this a bug since these arguments work in this library!

**Comment by jcs090218** [comment]
I have applied a fix in b1945507ef3f5ec1753cda36cd65250d6a8a2afe.

Can you check to see if this work for you?

**Comment by arminfriedl** [comment]
Thanks! `0.1.3` works without any issues for me now.
